code/game/machinery/dehumidifier.dm

```diff
@@ -0,0 +1,103 @@
+//This magic dehumidifier is a portable water-to-nothing machine, although it does need a charged cell to work.
+//See drain.dm!
+/obj/machinery/dehumidifier
+	name = "emergency dehumidifier"
+	desc = "A portable dehumidifier with emergency high-volume pump to help keep you dry when it's getting damp... just very slowly."
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "dhum-off"
+	anchored = FALSE
+	density = TRUE
+
+	construct_state = /decl/machine_construction/default/panel_closed
+	uncreated_component_parts = null
+	base_type = /obj/machinery/dehumidifier
+	stat_immune = 0
+
+	power_channel = LOCAL
+	use_power = POWER_USE_OFF
+
+	atom_flags = ATOM_FLAG_CLIMBABLE
+	clicksound = "switch"
+
+	var/active = FALSE
+	var/drainage = TRUE
+	var/last_gurgle = 0
+
+/obj/machinery/dehumidifier/Initialize()
+	. = ..()
+	queue_icon_update()
+
+/obj/machinery/dehumidifier/on_update_icon(rebuild_overlay = FALSE)
+	icon_state = "dhum-[active ? "on" : "off"]"
+	if(rebuild_overlay)
+		overlays.Cut()
+		if(panel_open)
+			overlays.Add("dhum-open")
+
+/obj/machinery/dehumidifier/examine(mob/user)
+	. = ..()
+	if(.)
+		to_chat(user, "The dehumidifier is [active ? "on" : "off"] and the hatch is [panel_open ? "open" : "closed"].")
+		var/obj/item/cell/cell = get_cell()
+		if(panel_open)
+			to_chat(user, "The power cell is [cell ? "installed" : "missing"].")
+		else
+			to_chat(user, "The charge meter reads [cell ? round(cell.percent(), 1) : 0]%")
+
+/obj/machinery/dehumidifier/proc/set_active(new_active)
+	if(active != new_active)
+		active = new_active
+		update_use_power(active ? POWER_USE_ACTIVE : POWER_USE_OFF)
+		update_icon()
```

Comment:
Maybe `queue` here as well, or possibly omit it. The above proc `update_use_power` also updates icon if a change is made. Also note that I'm not sure you actually _want_ to set use_power here; I don't think it does anything, since you aren't setting `active_power_usage`.